shell/browser/ui/electron_desktop_window_tree_host_linux.cc

```diff
@@ -251,13 +253,36 @@ void ElectronDesktopWindowTreeHostLinux::DispatchEvent(ui::Event* event) {
         is_mousedown &&
         (mouse_event->IsRightMouseButton() ||
          (mouse_event->IsLeftMouseButton() && mouse_event->IsControlDown()));
-    if (is_system_menu_trigger) {
-      electron::api::WebContents::SetDisableDraggableRegions(true);
+
+    if (!is_system_menu_trigger) {
       views::DesktopWindowTreeHostLinux::DispatchEvent(event);
-      electron::api::WebContents::SetDisableDraggableRegions(false);
       return;
     }
+
+    // Determine the non-client area and dispatch 'system-context-menu'.
+    ui::LocatedEvent* located_event = event->AsLocatedEvent();
+    int hit_test_code = HTNOWHERE;
+    if (GetContentWindow() && GetContentWindow()->delegate()) {
+      gfx::PointF location = located_event->location_f();
+      gfx::PointF location_in_dip =
+          GetRootTransform().InverseMapPoint(location).value_or(location);
+      hit_test_code = GetContentWindow()->delegate()->GetNonClientComponent(
+          gfx::ToRoundedPoint(location_in_dip));
+      if (hit_test_code != HTCLIENT && hit_test_code != HTNOWHERE) {
+        bool prevent_default = false;
+        native_window_view_->NotifyWindowSystemContextMenu(
+            located_event->x(), located_event->y(), &prevent_default);
+        if (prevent_default)
+          return;
+      }
```

Comment:
I believe this has to be tweaked a bit,

1) If the application wants to show a custom menu which is the `prevent_default` case then we should proceed to propagate the events so that `contextmenu` event ends up firing.

2) If the application wants the OS to draw the menu then we should make the following calls

```
if (ui::OzonePlatform::GetInstance()
                ->GetPlatformRuntimeProperties()
                .supports_server_window_menus) {
          desktop_window_tree_host_->ShowWindowControlsMenu(
              display::Screen::GetScreen()->GetCursorScreenPoint());
              ...
        }
```

Reply:
@deepak1556 just pushed a change that should address this